src/linear_algebra/ray.lean

```diff
@@ -30,162 +30,220 @@ variables {M : Type*} [add_comm_monoid M] [module R M]
 variables {N : Type*} [add_comm_monoid N] [module R N]
 variables (ι : Type*) [decidable_eq ι]
 
-/-- Two vectors are in the same ray if some positive multiples of them are equal (in the typical
-case over a field, this means each is a positive multiple of the other).  Over a field, this
-is equivalent to `mul_action.orbit_rel`. -/
+/-- Two vectors are in the same ray if either one of them is zero or some positive multiples of them
+are equal (in the typical case over a field, this means one of them is a nonnegative multiple of
+another). -/
 def same_ray (v₁ v₂ : M) : Prop :=
-∃ (r₁ r₂ : R), 0 < r₁ ∧ 0 < r₂ ∧ r₁ • v₁ = r₂ • v₂
+v₁ = 0 ∨ v₂ = 0 ∨ ∃ (r₁ r₂ : R), 0 < r₁ ∧ 0 < r₂ ∧ r₁ • v₁ = r₂ • v₂
```

Comment:
Another possible spelling would be
```suggestion
∃ (r₁ r₂ : R), 0 < (r₁, r₂) ∧ r₁ • v₁ = r₂ • v₂
```
Is that equivalent?

Reply:
Ah, it's not as it requires nontriviality of R for many results, so nevermind

Reply:
It is equivalent to my definition if `R` is nontrivial and we have `no_zero_smul_divisors R M`.